packages/contracts/contracts/ionic/vault/OptimizedAPRVaultFirstExtension.sol

```diff
@@ -0,0 +1,201 @@
+                            ADAPTER LOGIC
+    ------------------------------------------------------------*/
+
+  event NewAdaptersProposed(AdapterConfig[10] newAdapter, uint8 adaptersCount, uint256 timestamp);
+
+  /**
+   * @notice Propose a new adapter for this vault. Caller must be Owner.
+   * @param newAdapters A new ERC4626 that should be used as a yield adapter for this asset.
+   * @param newAdaptersCount Amount of new adapters.
+   */
+  function proposeAdapters(AdapterConfig[10] calldata newAdapters, uint8 newAdaptersCount) external onlyOwner {
+    _verifyAdapterConfig(newAdapters, newAdaptersCount);
+
+    for (uint8 i; i < newAdaptersCount; i++) {
+      proposedAdapters[i] = newAdapters[i];
+    }
+
+    proposedAdaptersCount = newAdaptersCount;
+
+    proposedAdapterTime = block.timestamp;
+
+    emit NewAdaptersProposed(newAdapters, proposedAdaptersCount, block.timestamp);
+  }
+
+  function getAllFlywheels() external view returns (IonicFlywheel[] memory allFlywheels) {
+    allFlywheels = new IonicFlywheel[](rewardTokens.length);
+    for (uint256 i = 0; i < rewardTokens.length; i++) {
+      allFlywheels[i] = flywheelForRewardToken[rewardTokens[i]];
+    }
+  }
+
+  /// @notice claim all token rewards
+  function claimRewards() public {
+    _claimRewards(msg.sender);
+  }
+
+  function claimRewardsForUser(address user) public {
+    _claimRewards(user);
+  }
+
+  function _claimRewards(address user) internal {
+    for (uint256 i = 0; i < rewardTokens.length; i++) {
+      IonicFlywheel flywheel = flywheelForRewardToken[rewardTokens[i]];
+      flywheel.accrue(ERC20(address(this)), user);
+      flywheel.claimRewards(user);
+    }
+  }
+
+  function _afterTokenTransfer(
+    address from,
+    address to,
+    uint256 amount
+  ) internal override {
+    super._afterTokenTransfer(from, to, amount);
+    for (uint256 i; i < rewardTokens.length; ++i) {
+      flywheelForRewardToken[rewardTokens[i]].accrue(ERC20(address(this)), from, to);
+    }
+  }
+
+  function addRewardToken(IERC20 token_) public {
+    require(msg.sender == owner() || msg.sender == address(this), "!owner or self");
+    require(address(flywheelForRewardToken[token_]) == address(0), "already added");
+
+    TransparentUpgradeableProxy proxy = new TransparentUpgradeableProxy(flywheelLogic, owner(), "");
+    IonicFlywheel newFlywheel = IonicFlywheel(address(proxy));
+
+    newFlywheel.initialize(
+      ERC20(address(token_)),
+      IFlywheelRewards(address(0)),
+      IFlywheelBooster(address(0)),
+      address(this)
+    );
+    FuseFlywheelDynamicRewards rewardsContract = new FuseFlywheelDynamicRewards(
+      FlywheelCore(address(newFlywheel)),
+      1 days
+    );
+    newFlywheel.setFlywheelRewards(IFlywheelRewards(address(rewardsContract)));
+    token_.approve(address(rewardsContract), type(uint256).max);
+    newFlywheel.updateFeeSettings(0, address(this));
+    // TODO accept owner
```

Comment:
check if the `// TODO` is implemented

Reply:
_setPendingOwner is a part of SafeOwnableUpgradeable which IonicFlywheelCore inherits, so we should be good

Reply:
I checked it and the owner is NOT changed to the admin address, it is the address of the vault until `_acceptOwner()` is called